atlas_metadata_validator/atlas_checker.py

```diff
@@ -93,12 +93,24 @@ def run_general_checks(self, logger):
         if not self.skip_file_checks:
             logger.info("Checking FASTQ URIs. This may take a while... (Skip this check with -x option)")
             uri_fields = get_controlled_vocabulary("raw_data_download_sdrf_fields")
+            uri_pattern = re.compile(r"^(https?|ftp)://", re.IGNORECASE)  # Identify based on start of string
+            non_uri_pattern = re.compile(r"^[^/\s]+$")  # Not allowing slashes and spaces if we don't have a web address
+            uris = []
+            non_uris = []
             for uri_field in uri_fields:
                 uri_index = [i for i, c in enumerate(self.sdrf_header)
                              if re.search(uri_field, c, flags=re.IGNORECASE)]
-                uris = [row[i] for row in self.sdrf for i in uri_index
-                        if re.match(re.compile("^ftp|^http", re.IGNORECASE), str(row[i]))]
-                # Using multi-threading to speed this up
+                for row in self.sdrf:
+                    for i in uri_index:
+                        if re.match(uri_pattern, str(row[i])):
+                            uris.append(row[i])
+                        elif not re.match(non_uri_pattern, row[i]):
+                            logger.error("{} {} is not in the right format.".format(uri_field.upper(), row[i]))
```

Comment:
Could this be made more explanatory for our future selves?

Reply:
Agree. I've re-organised this bit, one section per check. Additionally, the "skip" option now only skips the web requests part. The other checks are now always run.